### PR TITLE
6 support for scraping kubectl log from each pod on all containers and storing in the health check folder

### PR DESCRIPTION
**Pull request by markcurtis1970**
I haven't implemented this exactly in the way we discussed but its more isolated. however the container logs is called from the cluster.go so its collected before the logs from the actual pods and they each go into a directory like so:

```
20230620-162913-DDC
...
├── kubernetes
│   ├── container-logs
```

### CODE REVIEW

File: cmd/root/kubernetes/containers_test.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//  Copyright 2023 Dremio Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (rsvihladremio) [inline cmd/root/kubernetes/containers_test.go:21]: remove the double license

File: cmd/root/kubernetes/containers.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+// CommandExecutor is an interface that represents the functionality of executing commands.
+type CommandExecutor interface {
+	ExecuteCommand(command string, args ...string) ([]byte, error)
+}
+
+// DefaultCommandExecutor is the default implementation of the CommandExecutor interface.
+type DefaultCommandExecutor struct{}
+
+// ExecuteCommand executes the specified command and returns the output and any error.
+func (e *DefaultCommandExecutor) ExecuteCommand(command string, args ...string) ([]byte, error) {
```
Comment (rsvihladremio) [inline cmd/root/kubernetes/containers.go:39]: on all of this executecommand stuff, we already use the cli.Cli struct to execute everything, I am not sure why we need to reimplement the same functionality here.

File: cmd/root/kubernetes/containers.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+	}
+	// Convert []byte to bytes.Buffer
+	outputBuffer := bytes.NewBuffer(out)
+
+	// Convert bytes.Buffer to string
+	podNames := outputBuffer.String()
+	simplelog.Debugf("Pods found: %v", podNames)
+	pods = split(podNames, "\n")
+
+	return err, pods
```
Comment (rsvihladremio) [inline cmd/root/kubernetes/containers.go:97]: I would like to limit this to the pods that match from executors and coordinators, so we should use the already parsed host. 

Have this function be a part of clusterCollect functions
https://github.com/dremio/dremio-diagnostic-collector/blob/9531526fd3fb95892b48b3387a2a8c7a10c1fddb/cmd/root.go#L126
just change it to take a []string argument


And pass in that []string of hosts here https://github.com/dremio/dremio-diagnostic-collector/blob/9531526fd3fb95892b48b3387a2a8c7a10c1fddb/cmd/root/collection/collector.go#L119

File: cmd/root/kubernetes/containers.go [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+}
+
+func GetInitContainerLogs(executor CommandExecutor, podName, outputDir string) (err error) {
+	pod := strings.TrimPrefix(podName, "pod/")
```
Comment (rsvihladremio) [inline cmd/root/kubernetes/containers.go:101]: we should just get all container logs at once which one can do with a one liner

 kubectl get pod dremio-master-0 -n default -o jsonpath='{.spec.initContainers[*].name} {.spec.containers[*].name}'

File: cmd/root/kubernetes/containers_test.go [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+	err := GetContainerLogs(mockExecutor, "podName", t.TempDir())
+
+	// Assert that the error is nil (indicating success)
+	if err != nil {
+		t.Errorf("Expected no error, but got: %v", err)
+	}
+}
+
+// Helper function to check if two string slices are equal
+func equalStringSlices(a, b []string) bool {
```
Comment (rsvihladremio) [inline cmd/root/kubernetes/containers_test.go:145]: just use reflect.DeepEqual

File: cmd/root/kubernetes/kubernetes.go [hunk 6]
```diff
@@ -128,3 +128,12 @@ func (c *KubectlK8sActions) FindHosts(searchTerm string) (podName []string, err
 func (c *KubectlK8sActions) HelpText() string {
 	return "Make sure the labels and namespace you use actually correspond to your dremio pods: try something like 'ddc -n mynamespace --coordinator app=dremio-coordinator --executor app=dremio-executor'.  You can also run 'kubectl get pods --show-labels' to see what labels are available to use for your dremio pods"
 }
+
+func (c *KubectlK8sActions) CopyLogs(hostString string, isCoordinator bool, source, destination string) (out string, err error) {
```
Comment (rsvihladremio) [inline cmd/root/kubernetes/kubernetes.go:132]: * Not sure why we're double cleaning here as the c.cleanLocal is already doing the same as the strings.Replace
* I don't think we need the CopyLogs bit at all do we?

**Review by rsvihladremio — CHANGES_REQUESTED** [review]

File: cmd/root/kubernetes/containers_test.go (hunk 1 above)
Comment (markcurtis1970) [reply cmd/root/kubernetes/containers_test.go]: this file is now removed completely

File: cmd/root/kubernetes/containers.go (hunk 2 above)
Comment (markcurtis1970) [reply cmd/root/kubernetes/containers.go]: file now removed completely

**Comment by markcurtis1970** [comment]
Still have tests to write but much more simplified now. Keeping all log collecting functionality in `cluster.go`

**Pull request merged**